### PR TITLE
Make Jest tests domless

### PR DESCRIPTION
**Pull request by interactivellama**
This paves the way for this library to work in a DOM-less environments--not that you'd want to UI components to work completely there, but this PR allows snapshot testing of applications if the consuming dev desires. There still may be components that are not compatible with a DOM-less environment, but ones with snapshot tests are.

Last four commits on top of #1140 

- Remove ButtonStateful extends TooltipTrigger: Delete tooltip/trigger (This was the only application using it)
- Guard DOM functions with conditions: This allows components to be rendered in an environment without a DOM.
- Make jest environment not use JSDOM: JSDOM environment has a document, etc. but React test renderer does provide DOM nodes for instances with React.findDOMNode() which is used by react-onclickoutside. react-onclickoutside tests for a DOM which JSDOM has, so it runs but fails.
- Switch to deep render for DOM snapshots

### CODE REVIEW

File: components/menu-dropdown/menu-dropdown.jsx [hunk 1]
```diff
@@ -58,7 +58,7 @@ import { MENU_DROPDOWN, MENU_DROPDOWN_TRIGGER, LIST } from '../../utilities/cons
 // clicks even when those clicks are over areas that wouldn't normally fire
 // click or touch events (for example, iframes). A single overlay is shared
 // between all dropdowns in the app.
-const overlay = document.createElement('span');
+const overlay = typeof (document) !== 'undefined' ? document.createElement('span') : { style: {} };
```
Comment (cormacmccarthy) [inline components/menu-dropdown/menu-dropdown.jsx]: Seeing `typeof (document) !== 'undefined'` cached in a `const documentDefined` and used could be nice. Not a blocker at all.

File: components/popover/popover.jsx [hunk 2]
```diff
@@ -52,7 +52,7 @@ import { POPOVER } from '../../utilities/constants';
 // clicks even when those clicks are over areas that wouldn't normally fire
 // click or touch events (for example, iframes). A single overlay is shared
 // between all popovers in the app.
-const overlay = document.createElement('span');
+const overlay = typeof (document) !== 'undefined' ? document.createElement('span') : { style: {} };
```
Comment (cormacmccarthy) [inline components/popover/popover.jsx]: Same comment as above.

**Comment by cormacmccarthy** [comment]
How should I test this?

Running `npm run test` on this branch left me with 76+ errors in the terminal. I believe the following is the Jest portion:

```
> design-system-react@0.7.3 snapshot-test /Users/cmcculloh/projects/design-system-react
> jest

 PASS  tests/radio-group/radio-group.snapshot-test.jsx
 PASS  tests/radio-button-group/radio-button-group.snapshot-test.jsx
 PASS  tests/avatar/avatar.snapshot-test.jsx (6.689s)
 PASS  tests/icon/icon.snapshot-test.jsx (7.686s)
 PASS  tests/toast/toast.snapshot-test.jsx
 PASS  tests/alert/alert.snapshot-test.jsx
 PASS  tests/checkbox/checkbox.snapshot-test.jsx
 PASS  tests/navigation/navigation.snapshot-test.jsx
 PASS  tests/menu-picklist/menupicklist.snapshot-test.jsx
  ● Console

    console.warn utilities/warning/low-priority-warning.js:13
      Warning: [Design System React] SLDSMenuPicklist is a deprecated component. Bugfixes can be contributed, but new features and additional alignment with SLDS may be declined. Please use an read-only Combobox instead. It is more accessible and closer aligned to SLDS.

 PASS  tests/panel/filtering.snapshot-test.jsx
  ● Console

    console.warn utilities/warning/low-priority-warning.js:13
      Warning: [Design System React] SLDSPanel is a prototype. (a) Props may change within a minor release. (b) Web Content Accessibility Guidelines may not be met. (c) CSS imports may be required, since it is being added to SLDS.
    console.warn utilities/warning/low-priority-warning.js:13
      Warning: [Design System React] SLDSFilter is a prototype. (a) Props may change within a minor release. (b) Web Content Accessibility Guidelines may not be met. (c) CSS imports may be required, since it is being added to SLDS.

 PASS  tests/bread-crumb/bread-crumb.snapshot-test.jsx
 PASS  tests/filter/filter.snapshot-test.jsx (11.508s)
  ● Console

    console.warn utilities/warning/low-priority-warning.js:13
      Warning: [Design System React] SLDSFilter is a prototype. (a) Props may change within a minor release. (b) Web Content Accessibility Guidelines may not be met. (c) CSS imports may be required, since it is being added to SLDS.

 PASS  tests/date-picker/datepicker.snapshot-test.jsx (11.713s)
  ● Console

    console.warn utilities/warning/low-priority-warning.js:13
      Warning: [Design System React] SLDSMenuPicklist is a deprecated component. Bugfixes can be contributed, but new features and additional alignment with SLDS may be declined. Please use an read-only Combobox instead. It is more accessible and closer aligned to SLDS.

 PASS  tests/accordion/accordion.snapshot-test.jsx
 PASS  tests/button-group/button-group.snapshot-test.jsx
 PASS  tests/combobox/combobox.snapshot-test.jsx (12.797s)
 PASS  tests/data-table/data-table.snapshot-test.jsx

Test Suites: 17 passed, 17 total
Tests:       116 passed, 116 total
Snapshots:   116 passed, 116 total
Time:        13.383s
Ran all test suites.
```

I assume those warnings are expected?

File: components/menu-dropdown/menu-dropdown.jsx (hunk 1 above)
Comment (interactivellama) [reply components/menu-dropdown/menu-dropdown.jsx]: Good idea.

**Comment by interactivellama** [comment]
The warnings are fine. Errors are problems though.

**Pull request merged**